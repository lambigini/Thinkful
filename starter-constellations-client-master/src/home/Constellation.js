import React from "react";

function Constellation ({ constellation }) {
return (
  <div className="card col-12 col-md-4 rounded-0">
    <div className="card-body">
      <h5 className="card-title">
        {constellation.name} -{" "}
        <small className="text-muted">{constellation.meaning}</small>
      </h5>
      <p className="card-text">
        <strong>Stars with Planets: </strong>
        {constellation.starsWithPlanets}
      </p>
      <p className="card-text">
        <strong>Quadrant: </strong>
        {constellation.quadrant}
      </p>
    </div>
  </div>
);
}
export default Constellation;
