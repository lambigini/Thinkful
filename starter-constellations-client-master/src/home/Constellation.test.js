import React from "react";
import Constellation from "./Constellation";
import renderer from "react-test-renderer";

it("renders a constellation card when provided data", () => {
  const constellation = {
    id: "zb8QvVt",
    name: "Crater",
    meaning: "Cup",
    starsWithPlanets: 10,
    quadrant: "SQ2",
  };
  const component = renderer.create(
    <Constellation constellation={constellation} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
