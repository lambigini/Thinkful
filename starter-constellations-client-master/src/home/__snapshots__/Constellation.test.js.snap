// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a constellation card when provided data 1`] = `
<div
  className="card col-12 col-md-4 rounded-0"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Crater
       -
       
      <small
        className="text-muted"
      >
        Cup
      </small>
    </h5>
    <p
      className="card-text"
    >
      <strong>
        Stars with Planets: 
      </strong>
      10
    </p>
    <p
      className="card-text"
    >
      <strong>
        Quadrant: 
      </strong>
      SQ2
    </p>
  </div>
</div>
`;
